doc/is_partitioned_until.qbk

```diff
@@ -0,0 +1,67 @@
+[/ File is_partitioned_until.qbk]
+
+[section:is_partitioned_until is_partitioned_until ]
+
+[/license
+Copyright (c) 2017 Alexander Zaitsev
+
+Distributed under the Boost Software License, Version 1.0.
+(See accompanying file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+]
+
+The header file 'is_partitioned_until.hpp' contains two variants of a single algorithm, `is_partitioned_until`. The algorithm tests to see if a sequence is partitioned according to a predicate; in other words, all the items in the sequence that satisfy the predicate are at the beginning of the sequence.
+
+The routine `is_partitioned_until` takes a sequence and a predicate. It returns the first iterator 'it' in the sequence [first, last) for which is_partitioned(first, it, p) is false. Returns last if the entire sequence is partitioned.
```

Comment:
I think that inverting this description might be better.  Something like "It returns the last iterator `it` in the sequence `[begin, end)` for which the `is_partitioned` is true."   That way, it is clear that if the entire sequence is partitioned, it returns `end`, since that's the last iterator in the sequence.

You might also want to note that every sequence of length < 2 is partitioned (or maybe I'll put that into the `is_partitioned` docs)